common/src/main/java/com/wynntils/models/experience/ExperienceModel.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.experience;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.components.Models;
+import com.wynntils.mc.event.SetXpEvent;
+import com.wynntils.models.experience.event.ExperienceGainEvent;
+import com.wynntils.models.worlds.WorldStateModel;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import com.wynntils.models.worlds.type.WorldState;
+import com.wynntils.utils.mc.McUtils;
+import java.util.List;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ExperienceModel extends Model {
```

Comment:
I was thinking, we have consistently used "Xp" instead of "Experience" as name for this. But XpModel might sound a bit short. I would suggest using the "proper full name" of "Combat XP", so name this `CombatXpModel` (and also `CombatXpFunctions`). Would that be ok to you?